client/src/app/site/modules/site-wrapper/services/voting-banner.service.ts

```diff
@@ -86,15 +98,23 @@ export class VotingBannerService {
      */
     private getTextForPoll(poll: ViewPoll): string {
         const contentObject = poll.getContentObject();
-        if (poll.isMotionPoll) {
-            const motionTranslation = this.translate.instant(`Motion`);
-            const votingOpenedTranslation = this.translate.instant(`Voting opened`);
-            return `${motionTranslation} ${contentObject.getNumberOrTitle()}: ${votingOpenedTranslation}`;
-        } else if (poll.isAssignmentPoll) {
-            return `${contentObject.getTitle()}: ${this.translate.instant(`Ballot opened`)}`;
-        } else {
-            return this.translate.instant(`Voting opened`);
-        }
+        return (
+            contentObject?.getVotingText(text => this.translate.instant(text), poll) ??
+            this.translate.instant(`Voting opened`)
+        );
+    }
+
+    /**
+     * Returns for a given poll a url for the banner.
+     *
+     * @param poll The given poll.
+     *
+     * @returns A string containing the url.
+     */
+    private getUrlForPoll(poll: ViewPoll): string {
+        return this.operator.hasPerms(Permission.meetingCanSeeAutopilot)
+            ? `/${this.activeMeeting.meetingId}/autopilot/`
+            : poll.getContentObjectDetailStateURL();
```

Comment:
Please use the function `getDetailStateUrl` and remove the function `getContentObjectDetailStateUrl` from `ViewPoll`s